### PR TITLE
Parser for CTL formula

### PR DESCRIPTION
**Pull request by valentinnodan**

### CODE REVIEW

File: verifier/src/main/kotlin/itmo/verifier/formula/Formula.kt [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    var formula: CTLFormula
+): CTLFormula()
+
+data class AG(
+    var formula: CTLFormula
+): CTLFormula()
+
+data class EG(
+    var formula: CTLFormula
+): CTLFormula()
```
Comment (Mervap) [inline verifier/src/main/kotlin/itmo/verifier/formula/Formula.kt]: Do we need `var` here?

File: verifier/src/test/kotlin/itmo/verifier/FormulaParserTest.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        val formulaString = "1"
+
+        assertEquals(CTLGrammar.parseToEnd(formulaString), TRUE)
+    }
+
+    @Test
+    fun `and test`() {
+        val formulaString = "1 & 0"
+
+        assertEquals(CTLGrammar.parseToEnd(formulaString), Not(Or(Not(TRUE), Not(Not(TRUE)))))
```
Comment (Mervap) [inline verifier/src/test/kotlin/itmo/verifier/FormulaParserTest.kt:53]: Not sure, but probably things like `Not(Not(...))` will have a bad effect on performance at complex formula

**Review by Mervap — APPROVED** [review]

File: verifier/src/main/kotlin/itmo/verifier/formula/Formula.kt [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+data class Or(
+    val left: CTLFormula,
+    val right: CTLFormula
+): CTLFormula()
+
+data class EX(
+    var formula: CTLFormula
+): CTLFormula()
```
Comment (Mervap) [inline verifier/src/main/kotlin/itmo/verifier/formula/Formula.kt]: Do we need var here?

File: verifier/src/test/kotlin/itmo/verifier/FormulaParserTest.kt (hunk 2 above)
Comment (valentinnodan) [reply verifier/src/test/kotlin/itmo/verifier/FormulaParserTest.kt:53]: Yes, I also thought about this
Decided not to put this in grammar, it seems not really it's matter 
But I'll write the optimization function inside the CTLFormula class

File: verifier/src/main/kotlin/itmo/verifier/formula/Formula.kt (hunk 3 above)
Comment (valentinnodan) [reply verifier/src/main/kotlin/itmo/verifier/formula/Formula.kt]: No, fixed

**Review by Mervap — APPROVED** [review]

**Pull request merged**